service/evaluation/evaluation_test.go

```diff
@@ -0,0 +1,2145 @@
+						assert.NoError(t, s.Create(orchestratortest.NewCatalog()))
+					})))},
+				},
+			},
+			want: nil,
+			wantErr: func(tt assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorContains(t, err, "invalid request: invalid GetControlRequest.CatalogId: value length must be at least 1 runes")
+			},
+		},
+		{
+			name: "control_id is missing",
+			fields: fields{
+				storage: testutil.NewInMemoryStorage(t),
+				orchestratorAddress: grpcTarget{
+					opts: []grpc.DialOption{grpc.WithContextDialer(newBufConnDialer(testutil.NewInMemoryStorage(t, func(s persistence.Storage) {
+						assert.NoError(t, s.Create(orchestratortest.NewCatalog()))
+					})))},
+				},
+			},
+			want: nil,
+			wantErr: func(tt assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorContains(t, err, "invalid request: invalid GetControlRequest.CatalogId: value length must be at least 1 runes")
+			},
+		},
+		{
+			name: "control does not exist",
+			fields: fields{
+				storage: testutil.NewInMemoryStorage(t),
+				orchestratorAddress: grpcTarget{
+					opts: []grpc.DialOption{grpc.WithContextDialer(newBufConnDialer(testutil.NewInMemoryStorage(t, func(s persistence.Storage) {
+						assert.NoError(t, s.Create(orchestratortest.NewCatalog()))
+					})))},
+				},
+			},
+			args: args{
+				catalogId:    "wrong_catalog_id",
+				categoryName: "wrong_category_id",
+				controlId:    "wrong_control_id",
+			},
+			want: nil,
+			wantErr: func(tt assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorContains(t, err, "control not found")
+			},
+		},
+		{
+			name: "Happy path",
+			fields: fields{
+				storage: testutil.NewInMemoryStorage(t),
+				orchestratorAddress: grpcTarget{
+					opts: []grpc.DialOption{grpc.WithContextDialer(newBufConnDialer(testutil.NewInMemoryStorage(t, func(s persistence.Storage) {
+						assert.NoError(t, s.Create(orchestratortest.NewCatalog()))
+					})))},
+				},
+			},
+			args: args{
+				catalogId:    testdata.MockCatalogID,
+				categoryName: testdata.MockCategoryName,
+				controlId:    testdata.MockControlID1,
+			},
+			want: func(tt assert.TestingT, i1 interface{}, i2 ...interface{}) bool {
+				gotControl, ok := i1.(*orchestrator.Control)
+				if !assert.True(tt, ok) {
+					return false
+				}
+
+				// We need to truncate the metric from the control because the control is only returned with its sub-control but without the sub-control's metric.
+				wantControl := orchestratortest.MockControl1
+				tmpMetrics := wantControl.Controls[0].Metrics
+				wantControl.Controls[0].Metrics = nil
+
+				if !proto.Equal(gotControl, wantControl) {
+					t.Errorf("Service.GetControl() = %v, want %v", gotControl, wantControl)
+					wantControl.Controls[0].Metrics = tmpMetrics
+					return false
+				}
+
+				wantControl.Controls[0].Metrics = tmpMetrics
+				return true
```

Comment:
Does anyone have an idea on how to do this better?

Reply:
I am not quite sure what you want to achieve?

Reply:
For the test I have to delete the Metrics. And if I do the following, the orchestratortest.MockControl1 is changed for all following tests and some tests fail. 
`
wantControl := orchestratortest.MockControl1
tmpMetrics := wantControl.Controls[0].Metrics
wantControl.Controls[0].Metrics = nil
`

I have no idea how to copy orchestratortest.MockControl1 without changing it permanently.